### ISSUE TITLE
console.log() always outputs a logging prefix on Windows

### ISSUE THREAD

**Issue by joaomoreno** [atom/electron#1846]
Consider the following script:

```javascript
console.log('hello world');
require('app').quit();
```

Running this on OS X behaves normally. The application exits successfully with the following output:

```
hello world
```

On Windows I get the following output:

```
[2648:0602/172546:INFO:CONSOLE(0)] hello world
```

This is unexpected and I've found no way to disable it.

In essence, I'm trying to execute Electron's executable as a replacement for Node. Are there any obvious issues that I will encounter, besides this one?

**Comment by joaomoreno** [comment]
On a related note, why not provide a way to simply run `node` without any overhead? Could be done with a command line argument to atom.exe or simply another .exe which binds to the right things.

**Comment by baconface** [comment]
Are you trying to make a command line application? Why not just drop Electron, install node.js and call it from command line?

**Comment by joaomoreno** [comment]
Not at all. My app is indeed Electron based.

In it I spawn processes which themselves spawn other processes and read their output. I own these second-level processes, I write them in whatever tool I have at my disposal. But I do not own the ones I spawn directly: I can only tell them which sub-processes to spawn.

I've been telling them "Hey, just spawn atom.exe with these arguments", which works great on OS X. Not so much on Windows do the the output prefixing.

**Comment by paulcbetts** [comment]
@joaomoreno How are you running these processes? Are you using `child_process.fork`?

**Comment by joaomoreno** [comment]
I use `cp.spawn` since what I'm spawning is not Javascript based; it's actual executables. They themselves need to spawn something that returns a result via `stdout`. I would like to use Javascript for that.

**Comment by zcbenz** [comment]
On Windows Node.js is a console program while Electron is a GUI program, and by default you can not see the console output of a GUI program. In Electron `console.log` is redirected to Chromium's logging system to make sure things can be printed, if you don't want this behavior you can override it back:

```js
console.log = require('console').log
```

**Comment by joaomoreno** [comment]
@zcbenz That doesn't seem to help.

```javascript
console.log('hello');
require('console').log('world');
require('app').quit();
```

Still ends up in:

```
[1984:0604/082931:INFO:CONSOLE(0)] hello
[1984:0604/082931:INFO:CONSOLE(0)] world
```

**Comment by zcbenz** [comment]
Hmm, you can try `process.stdout.write` which Node's `console.log` is implemented with, but it might not work in a GUI program.

**Comment by joaomoreno** [comment]
@zcbenz It does work, but unfortunately ends up with the same prefix. All of them seem to be patched.

**Comment by zcbenz** [comment]
Sorry I forgot I had patched all of them, currently you have to create the stream by copying Node's code:
https://github.com/nodejs/io.js/blob/master/src/node.js#L616-L629

But unless you run Electron with stdout attached, like using `child_process.spawn`, writing to stdout would throw exception.

**Comment by etiktin** [comment]
This was fixed in v0.33.7
